rustworkx-core/src/centrality.rs

```diff
@@ -335,6 +335,48 @@ fn accumulate_edges<G>(
         }
     }
 }
+/// Compute the degree centrality of all nodes in a graph.
+///
+/// The algorithm used in this function is based on:
+///
+/// Degree Centrality, Betweenness Centrality, and Closeness Centrality in Social Network
+/// Atlantis Press - Junlong Zhang, Yu Luo 
+/// https://doi.org/10.2991/msam-17.2017.68 
+///
+/// Example usage of `degree_centrality` function
+/// ```
+/// use rustworkx_core::petgraph::graph::UnGraph;
+/// use rustworkx_core::centrality::degree_centrality;
+///
+/// // Create an undirected graph with 4 nodes and 4 edges
+/// let graph = UnGraph::<i32, ()>::from_edges(&[
+///     (0, 1), (1, 2), (2, 3), (3, 0)
+/// ]);
+///
+/// // Calculate degree centrality
+/// let centrality = degree_centrality(&graph);
+///
+/// // Each node has 2 connections out of 3 possible (n-1 where n=4)
+/// for value in centrality {
+///     assert!((value - 2.0/3.0).abs() < 1e-10);
+/// }
+/// ```
+
+
+pub fn degree_centrality<G>(graph: G) -> Vec<f64>
+where
+    G: IntoNodeIdentifiers + NodeCount + IntoNeighbors + NodeIndexable,
+{
+    let node_count = graph.node_count() as f64;
+    let mut centrality = vec![0.0; graph.node_bound()];
+
+    for node in graph.node_identifiers() {
+        let degree = graph.neighbors(node).count() as f64;
```

Comment:
If you want to parametrize this to support `in_degree` and `out_degree`, call https://docs.rs/petgraph/latest/petgraph/graph/struct.Graph.html#method.neighbors_directed and accept an `Option<Directed>` https://docs.rs/petgraph/latest/petgraph/enum.Direction.html with `None, Some(), Some(Outgoing), Some(Incoming)` instead?